### PR TITLE
Refactor: height responsiveness

### PR DESCRIPTION
**Pull request by pdpino**
* When `hoursInDisplay` changes, the UI updates smoothly (before it jumped, or did not update at all)
* Responsive to orientation changes

https://user-images.githubusercontent.com/20805451/177889593-dfed2c39-9fbd-48ea-9da6-07c53afa37e7.mp4

This is needed for the coming pinch-vertical feature (#130)

### CODE REVIEW

**Review by NiharR27 — APPROVED** [review]

File: src/Header/Header.js [hunk 1]
```diff
@@ -70,21 +61,19 @@ const WeekViewHeader = ({
...
-            column={column}
-            numberOfDays={numberOfDays}
-            format={formatDate}
-            DayComponent={DayComponent}
-            TodayComponent={TodayComponent}
-            onDayPress={onDayPress}
-            width={dayWidth}
-          />
-        ))}
+      {(columns || []).map((column) => (
```
Comment (hoangnm) [inline src/Header/Header.js:64]: I think it's nicer to use `||` on line 61.

File: src/utils/dimensions.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+/**
+ * Convert period of time (in minutes) to height (pixels in y dim).
+ *
+ * @param {*} minutesDelta period of time in minutes
+ * @param {*} verticalResolution resolution in minutes
+ * @returns pixels
+ */
+export const minutesToHeight = (minutesDelta, verticalResolution) => {
+  'worklet';
```
Comment (hoangnm) [inline src/utils/dimensions.js:68]: could you explain this line?

File: src/Header/Header.styles.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import { StyleSheet } from 'react-native';
+import { HEADER_HEIGHT } from '../utils/dimensions';
 
 const styles = StyleSheet.create({
   container: {
     flex: 1,
     flexDirection: 'row',
-    height: 50,
+    height: HEADER_HEIGHT,
```
Comment (hoangnm) [inline src/Header/Header.styles.js:8]: Do you think the header's height can be changed by dimension too?

Comment (pdpino) [reply src/Header/Header.styles.js:8]: Not for now, I think is ok to be fixed at 50

File: src/utils/dimensions.js (hunk 2 above)
Comment (pdpino) [reply src/utils/dimensions.js:68]: This line declares the function as a [rn-reanimated worklet](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/worklets). In summary, it means its code can be run in the UI thread --> more smooth animations

(I added some more context in the file docstring)

Comment (hoangnm) [reply src/utils/dimensions.js:72]: I just read from the doc, it says to run on UI thread, we must put the function to `runOnUI` function. Am I missing anything?

Comment (pdpino) [reply src/utils/dimensions.js]: Also run on the UI if is called from a reanimated hook, like `useDerivedValue`, `useAnimatedStyle`, etc. (I just know this from experience, not from the docs: if I try to use a non-`'worklet'` function with a hook --> throws error asking for a worklet)

Comment (pdpino) [reply src/utils/dimensions.js]: I just noticed this:

* `minutesInDayToTop()` is called from a `useDerivedValue()` function --> needs to be a worklet
* `minutesInHeight()` does not need to be a worklet

I removed the worklet from the latter

**Comment by hoangnm** [comment]
Great PR!

**Pull request merged**